### ISSUE TITLE
WON and conflated legacy data imports

### ISSUE THREAD

**Issue by geohacker** [orma/openroads-vn-analytics#473]
This ticket captures instruction to support legacy vpromm and WON imports by @PCDuc. Legacy vpromm data exists in Thanh Hoa which needs to be merged with WON data before we can import into ORMA.

Once the conflated WON is prepared, the instructions to upload new WON data is here https://github.com/orma/openroads-vn-api#whole-of-network-import. This involves:
* cloning the repo `git clone https://github.com/orma/openroads-vn-api.git`
* change to utils directory `cd utils`
* run the script `./won-import.sh data.shp uid username http://orma-api.url`

I will assist with the username and API URLs. Please let me know if there are any questions!

**Comment by geohacker** [comment]
List of data assets shared by @PCDuc - https://docs.google.com/document/d/12Zle0swGh1QWJ2SaJurql2VyKYHwyqoMKg8RK9mBWXQ/edit

**Comment by PCDuc** [comment]
This is the link to access all data listed in the document. Since I'll work on these data, please confirm the status of data with me before using them: https://1drv.ms/f/s!AlXoorZjtVf-ul__34h2sZpewW7e

**Comment by geohacker** [comment]
@PCDuc thank you, yes will confirm with you before using any.

Could you also define a timeline for the above plan?

**Comment by PCDuc** [comment]
@geohacker All data will be check and ready by Oct 12 except for Binh Dinh one. It will take me few days. I've confirmed with Binh Dinh to stop importing data and will finish the extract of Binh Dinh data by Oct 12 and the conflation by Oct 17. We will have time to do the removal and importing of data after Oct 12 and final import for Binh Dinh on Oct 18. Does it work for you?

**Comment by geohacker** [comment]
@PCDuc Could you help with the following items:

* VPROMM CSV in the format as previously uploaded to ORMA
* Is the WON data in this folder final https://onedrive.live.com/?authkey=%21AP_fiHaxml7Bbt4&id=FE57B563B6A2E855%217522&cid=FE57B563B6A2E855 - including Thanh Hoa? 
* I know that you wanted to consolidate Thanh Hoa vpromms but also remove potential conflicts from the Cam Thuy, Thac Than, and Lao Cai. Could you confirm please?

**Comment by PCDuc** [comment]
@geohacker 
> * VPROMM CSV in the format as previously uploaded to ORMA
Do you mean that you need the data to be standardize at Section level?. I've shared you VPRoMMS excel file at Road level before (see this link: https://1drv.ms/f/s!AlXoorZjtVf-umHru-LugUQlXCSB).
> * Is the WON data in this folder final https://onedrive.live.com/?authkey=%21AP_fiHaxml7Bbt4&id=FE57B563B6A2E855%217522&cid=FE57B563B6A2E855 - including Thanh Hoa?
WON for Thanh Hoa was done and put in the above folder. However, I still need some more time to work on the conflation for Binh Dinh data.
> * I know that you wanted to consolidate Thanh Hoa vpromms but also remove potential conflicts from the Cam Thuy, Thac Than, and Lao Cai. Could you confirm please?
Not sure the question. But the WON for 3 district (Cam Thuy, Thach Thanh, and Yen Dinh) have been merged with Thanh Hoa Vietbando layer to create one single WON layer (all conflicts have been removed).

**Comment by PCDuc** [comment]
@geohacker  WON for Binh Dinh is ready now in the shared folder: https://1drv.ms/f/s!AlXoorZjtVf-umPQnKfSOrg2jGM_

**Comment by geohacker** [comment]
@PCDuc great, thank you! Do you want to start importing new WONs to test? I'll clear out the database for Binh Dinh and Thanh Hoa in the mean time.

**Comment by PCDuc** [comment]
@geohacker Sound great! Let's start from Thursday afternoon

**Comment by geohacker** [comment]
@PCDuc can I import the VPROMM table that's in the drive https://onedrive.live.com/view.aspx?cid=fe57b563b6a2e855&page=view&resid=FE57B563B6A2E855!8876&parId=FE57B563B6A2E855!7521&authkey=!AOu74u6BRCVcJIE&app=Excel `VPROMMS 14 provinces 2018 Clean (20181025)`?

Please confirm.

**Comment by olafveerman** [comment]
Feedback:

DBC: 
> Again, I still would like to raise my questions:
> Is there any criteria for "clean data"?
> Moreover, we should concern about the future case when PDoT users have VPROMMs dataset and want to import it into ORMA. Is it easy for them to import without anyone supporting them to clean data? Or in other words, is there any tool in ORMA for easily importing VPROMMs dataset?

cc @geohacker

**Comment by geohacker** [comment]
> Is there any criteria for "clean data"?

This is a question for @PCDuc because he has the best knowledge of data sources and management. 

>  Is it easy for them to import without anyone supporting them to clean data? Or in other words, is there any tool in ORMA for easily importing VPROMMs dataset?

All imports can be performed by anybody but they require some technical understanding. Adding new VPROMM can be done via the Asset page itself. Importing new WON will require knowledge of running commands. This is part of the [documentation](https://github.com/orma/openroads-vn-api#whole-of-network-import) but not recommended to be done by PDoT.

**Comment by geohacker** [comment]
Noting that @PCDuc needs more time for preparing the changes to VPROMM table and imports are currently waiting for this. Waiting for a time estimate.

**Comment by PCDuc** [comment]
@geohacker VPRoMMS table is good to go now (https://1drv.ms/f/s!AlXoorZjtVf-umHru-LugUQlXCSB)
Please note that this is the VPRoMMS 2017 table at road level (only includes road attribute)

**Comment by geohacker** [comment]
@PCDuc thank you. I'm starting to import these but looks like there are duplicate entries for road ids in the table. Could help with this please? For example: 674ML00011, 674ML00014

**Comment by PCDuc** [comment]
@geohacker Thank you for pointing it out. I've found 3 duplicates and fixed them in the shared file.

**Comment by geohacker** [comment]
@PCDuc thank you for the quick fix. As we discussed just now on the phone, another new problem I'm seeing is this - there are road IDs missing in the vpromm table, that was previously added to ORMA and have geometries. For example `213YD00013`. 

If you could reconcile the vpromm table with the WON and legacy data, that will be the most ideal. Otherwise we may miss some data that were collected previously.

**Comment by PCDuc** [comment]
@geohacker Just double check the data, the VPRoMMS table we have here has already covered the list of GPRoMMS for 3 districts in Thanh Hoa WON. The example you raised is the one had been removed due to duplicate geometry with 213YD00012. So, you can go forward and import the VPRoMMS table.
Please note that we are still losing some data on section attributes because we only import VPRoMMS table at road level.

**Comment by geohacker** [comment]
@PCDuc we're in the process of importing the WON layers now. So far I've processed Thanh Hoa, Quang Nam, Quang Binh, Binh Dinh, and Lao Cai - all [from this folder](https://onedrive.live.com/?authkey=%21AP_fiHaxml7Bbt4&id=FE57B563B6A2E855%217522&cid=FE57B563B6A2E855)

I'm now looking at the folder ["WON layers for 10 provinces"](https://onedrive.live.com/?authkey=%21AP_fiHaxml7Bbt4&id=FE57B563B6A2E855%217528&cid=FE57B563B6A2E855) and the OSM shapefile overlaps with all the above provinces. Do you want me to import this shapefile? If we import, note that this will create duplicates and overlapping geometries in the above listed provinces. Please confirm. Thank you!

**Comment by PCDuc** [comment]
@geohacker Sorry, I should have uploaded the clipped layer for 10 provinces that I did in last August. 
I've downloaded the latest OSM layer and clipped it again for 10 provinces. Please find in the shared folder  "WON layers for 10 provinces".

**Comment by geohacker** [comment]
Thank you for fixing that @PCDuc.

All WON layers are now imported.
